### PR TITLE
#444 create relationships

### PR DESCRIPTION
**Pull request by laritakr**
# Story

This creates a service to create relationships between works and collections based on a spreadsheet. A separate job is submitted for each relationship.

Call on rails console... switch to 'sdapi' tenant, then run `AddWorksToCollections.call`
Refs: 

- #444

# Expected Behavior Before Changes

# Expected Behavior After Changes

# Screenshots / Video

<details>
<summary></summary>

</details>

# Notes

### CODE REVIEW

**Comment by ShanaLMoore** [comment]
We will need to clean up this branch if we merge it into rerun-errored-entries... 

It includes work that hasn't been QA'd or approved by the client. We will need this as a one off solution in production, without the additional work included unless Katharine approves them all by then (ie: snippets).

**Comment by laritakr** [comment]
> Thanks for taking this on and trying out a different approach! We will need to clean up this branch if we merge it into [rerun-errored-entries-with-uplimit](https://github.com/scientist-softserv/adventist-dl/tree/rerun-errored-entries-with-uplimit)
> 
> It includes work that hasn't been QA'd or approved by the client. We will need this as a one off solution in production, without the additional work included unless Katharine approves them all by then (ie: snippets).

We had to have the solr URL fix in... that's why I based it off of main. Changing the PR to go over main so you can see the 3 files included. We can easily pull them into whatever branch you need them in.

File: app/jobs/add_works_to_collections_job.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+      collection.try(:reindex_extent=, Hyrax::Adapters::NestingIndexAdapter::LIMITED_REINDEX)
+      work.member_of_collections << collection
+      work.save!
+      Rails.logger.info("🦄🪺👩‍👩‍👧‍👦 #{record_data[:identifier]} processed")
+    rescue StandardError => e
+      notice = "😈😈😈 ERROR: unable to create relationship for #{record_data[:identifier]}"
+      Rails.logger.error(notice)
+      Rails.logger.error(e.message)
+      File.open('lib/data/add_works_to_collection_errors.txt', 'a') do |f|
```
Comment (ShanaLMoore) [inline app/jobs/add_works_to_collections_job.rb:19]: I think if you were to save this to the public directory, an endpoint would be available for easy download. Fact check me though 😅

**Comment by ShanaLMoore** [comment]
For the record, I am going to cherry-pick the changes here into the rerun branch to avoid the changes from main.

File: app/jobs/add_works_to_collections_job.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+class AddWorksToCollectionsJob < ApplicationJob
+  queue_as :relationships
+
+  def perform(record_data:)
+    begin
```
Comment (ShanaLMoore) [inline app/jobs/add_works_to_collections_job.rb:7]: You don't need this begin here.

**Pull request closed without merging**